### PR TITLE
Support for a client joining after the simulation has started

### PR DESCRIPTION
**Pull request by eundersander**
## Motivation and Context

This is for the case that the server starts with a client, then that client drops and a new client joins (or the same client re-joins).

Also some unrelated polish in a separate commit here.

## How Has This Been Tested

Local testing on Macbook and Unity Editor client

## Types of changes

PR into demo branch

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

File: examples/siro_sandbox/app_states/app_state_fetch.py [hunk 1]
```diff
@@ -431,10 +430,10 @@ def _update_grasping_and_set_act_hints_local(self):
                 self.state_machine_agent_ctrl.current_state = FetchState.BRING
 
         # Do vis on the real navigation target
-        if self.state_machine_agent_ctrl.current_state != FetchState.WAIT:
-            safe_pos = self.state_machine_agent_ctrl.safe_pos
-            if safe_pos is not None:
-                self._draw_circle(safe_pos, color=mn.Color3.red(), radius=0.25)
+        # if self.state_machine_agent_ctrl.current_state != FetchState.WAIT:
```
Comment (eundersander) [inline examples/siro_sandbox/app_states/app_state_fetch.py:433]: Disabling this viz as it isn't needed for the demo

File: examples/siro_sandbox/app_states/app_state_fetch.py [hunk 2]
```diff
@@ -334,24 +334,23 @@ def _update_grasping_and_set_act_hints_local(self):
...
-                                throw_obj_id
-                            )
-                        )
                         throw_vel = (
                             self._throw_helper.viz_and_get_humanoid_throw()
                         )
-                        if not disable_spot:
-                            self.state_machine_agent_ctrl.current_state = (
-                                FetchState.SEARCH
+                        if throw_vel:
```
Comment (eundersander) [inline examples/siro_sandbox/app_states/app_state_fetch.py:340]: unrelated bug fix: throw_vel can be None depending on the server window camera

File: examples/siro_sandbox/app_states/app_state_fetch.py [hunk 3]
```diff
@@ -642,14 +641,14 @@ def _get_status_text(self):
         fetch_state_names = {
             # FetchState.WAIT : "",
             FetchState.SEARCH: "searching for object",
-            FetchState.SEARCH_ORACLE_NAV: "searching for object",
+            FetchState.SEARCH_ORACLE_NAV: "oracle nav to object",
```
Comment (eundersander) [inline examples/siro_sandbox/app_states/app_state_fetch.py:644]: unrelated GUI text polish

File: examples/siro_sandbox/server/keyframe_utils.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any
+
+
+def update_consolidated_keyframe(consolidated_keyframe, inc_keyframe):
```
Comment (eundersander) [inline examples/siro_sandbox/server/keyframe_utils.py:10]: I don't like this whole approach of doing detailed gfx-replay keyframe logic in Python, in the HITL codebase. This is hacky. Hopefully we'll clean up the HITL codebase in the future.

**Review by eundersander — COMMENTED** [review]
Some comments for reviewers.

**Review by 0mdc — APPROVED** [review]
LGTM! I'll try this out.
